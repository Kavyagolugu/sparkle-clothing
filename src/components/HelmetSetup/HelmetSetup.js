import React from 'react';
import { HelmetProvider, Helmet } from 'react-helmet-async';
import PropTypes from 'prop-types';

// helmet setup for the title
const HelmetSetup = ({ title }) => {
  return (
    <HelmetProvider>
      <Helmet>
        {/* title to be displayed on the browser */}
        <title>{title}</title>
        {/* <link rel="icon" href={href} /> */}
      </Helmet>
    </HelmetProvider>
  );
};

// prop type validation for helmet setup
HelmetSetup.propTypes = {
  title: PropTypes.string
};

export default HelmetSetup;
