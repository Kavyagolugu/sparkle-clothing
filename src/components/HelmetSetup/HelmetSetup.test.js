import React from 'react';
import renderer from 'react-test-renderer'
import HelmetSetup from './HelmetSetup';

describe('HelmetSetup', () => {
  const props = {
    title: 'Test Title',
    href: 'https://example.com/favicon.ico',
    metaDesc: 'Test Description'
  };

  // testing helmet setup props
  it('renders correctly', () => {
    const tree = renderer
      .create(<HelmetSetup {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
