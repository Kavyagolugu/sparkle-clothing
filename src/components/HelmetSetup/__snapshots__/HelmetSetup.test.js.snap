// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelmetSetup renders correctly 1`] = `null`;
